f2/src/forms/MidFeebackForm.js

```diff
@@ -0,0 +1,151 @@
+/** @jsx jsx */
+import React from 'react'
+import { jsx } from '@emotion/core'
+import { H1 } from '../components/header'
+import { useLingui } from '@lingui/react'
+import { Trans } from '@lingui/macro'
+import { Form, Field, useField } from 'react-final-form'
+import { Stack, FormControl, Box } from '@chakra-ui/core'
+import { FormLabel } from '../components/FormLabel'
+import { FormHelperText } from '../components/FormHelperText'
+import { Checkbox } from '../components/checkbox'
+import { Button } from '../components/button'
+import { Layout } from '../components/layout'
+import { TextArea } from '../components/text-area'
+
+const Control = ({ name, ...rest }) => {
+  const {
+    meta: { error, touched },
+  } = useField(name, { subscription: { touched: true, error: true } })
+  return <FormControl {...rest} isInvalid={error && touched} />
+}
+
+const CheckboxArrayControl = ({ name, value, defaultIsChecked, children }) => {
+  const {
+    input: { checked, ...input },
+    meta: { error, touched },
+  } = useField(name, {
+    type: 'checkbox', // important for RFF to manage the checked prop
+    value, // important for RFF to manage list of strings
+    defaultIsChecked,
+  })
+
+  return (
+    <Checkbox {...input} isChecked={checked} isInvalid={error && touched}>
+      {children}
+    </Checkbox>
+  )
+}
+
+export const MidFeedbackForm = ({ onSubmit }) => {
```

Comment:
could you add a short test for this form?

Reply:
Added. Also fixed the issue with the button not submitting: it was outside the form.... 😬